src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATE_HISTORY = [];
+  const CLONED_STATE = { ...state };
+
+  for (const ACTION of actions) {
+    switch (ACTION.type) {
+      case 'addProperties':
+        for (const KEY in ACTION.extraData) {
+          CLONED_STATE[KEY] = ACTION.extraData[KEY];
+        }
+
+        STATE_HISTORY.push({ ...CLONED_STATE });
```

Comment:
Pushing the cloned state into the history array should be done once after the switch statement to adhere to the DRY principle.